internal/elasticsearch/v8/client.go

```diff
@@ -25,7 +25,7 @@ import (
 	"github.com/elastic/go-elasticsearch/v8"
 )
 
-func NewClient(cfg interface{}) (*Client, error) {
+func NewClient(cfg interface{}, indexFn func(opencdc.Record) (string, error)) (*Client, error) {
 	configTyped, ok := cfg.(config)
```

Comment:
I would think about expanding the `config` interface with the method `IndexFunction() (f IndexFn, err error)`, although the `IndexFn` type would currently create a circular dependency, so it would have to be declared somewhere else. Anyway, just a thought, I'll let you decide if you want to do that, depending on how hard it would be to implement.

Also a sidenote - the fact that these clients are using `interface{}` extensively instead of the actual types is a bit of a code smell. I assume this was done so that all clients implement the same [interface](https://github.com/conduitio-labs/conduit-connector-elasticsearch/blob/5cdf17c71d542599aaac7afec7f817f5118585ed/internal/elasticsearch/client.go#L25), but I wonder if there's a better solution. One solution that comes to mind is to create a package like `internal/elasticsearch/common` and place the common interfaces there (for `Config`, `Payload`, `Metadata`). However, **this is not the point of this PR, so there is no need to do this right now**, I just noticed it and had to mention it 🙃

Reply:
Okay, now I have used `GetIndex(r opencdc.Record) (string, error)` method on the config interface only to determine the static or templated index.